### ISSUE TITLE
Declare `session.defaultParams` and `session.defaultOptions` stable

### ISSUE THREAD

**Issue by lucaswerkmeister** [lucaswerkmeister/m3api#21]
Proposed in https://github.com/lucaswerkmeister/m3api/issues/6#issuecomment-974697411, needed for other packages as well (cf. #20).

**Comment by lucaswerkmeister** [comment]
There’s also a `DEFAULT_OPTIONS` in `core.js`, which are the compile-time default options when neither request nor session default options specify a certain option. Should we export that as well? Then other packages could also use it…

And if we export it – how much do we lock it down? Do we encourage other packages to add their default options to it? (Which would already rule out `Object.freeze()` and `Object.seal()`.) If yes, do we try to protect our own options (e.g. make them non-configurable or non-writable)?

I think I’m leaning towards exporting these options indeed, and encouraging other packages to add their own default options to it, and strongly discouraging the modification of existing default properties, but without trying to prevent that on a technical level.

**Comment by lucaswerkmeister** [comment]
(I have no idea what I meant by “compile-time” up there btw. It’s JavaScript.)

**Comment by lucaswerkmeister** [comment]
Note: once we allow custom options, `combine.js` will have to compare the options differently than it does at the moment, because `JSON.stringify()` silently discards functions (compare 12989709daccf17a5e8e29a0823b73835e7d252a). Also, modules should take care to pass through unknown options instead of dropping them when they destructure an `options` object into known options (this is relevant in some code I’m currently writing for #6).

**Comment by lucaswerkmeister** [comment]
I think this is done; at some point I’ll add guidelines for extension packages somewhere (including how they should use the options), but that can be part of another issue (maybe #20).

**Comment by lucaswerkmeister** [comment]
Hm, one thing I didn’t realize until just now is that it’s not obvious how extension packages should import `DEFAULT_OPTIONS`. Packages that directly depend on m3api could use `./node_modules/m3api/core.js`, but for packages that have it as a *peer* dependency, that doesn’t work.

I think the only viable option is `m3api/core.js`, which works on Node; people using the packages for the browser will have to configure an import map or use a bundler :/
